### ISSUE TITLE
Player sprite sheet doesn't revert to default when changing to a new scene

### ISSUE THREAD

**Issue by m-masaa** [chrismaltby/gb-studio#458]
**Describe the bug**
As of version 2.0, when "Set Player Sprite Sheet" is used to modify the player character, the change is kept in subsequent scenes.
This is problematic when changing the player to a cursor in a Menu for instance as it will remain that way once exiting it and returning in-game.

**To Reproduce**
1. Set a Sprite Sheet to the player using the "Set Player Sprite Sheet" event. (Sprite Sheet must be different than the default)
2. Switch to a different scene
3. See error

**Expected behavior**
The starting Player Sprite Sheet should be used when changing to a new scene.

**Comment by RichardULZ** [comment]
I don't believe this was a thing in 1.2.1
This would break a lot of uses where someone wants to use a scene to change the character you play as.

Just set it to the correct sprite on your way out of the menu scene, i think we only store/restore the active player sprite for save/load events.

**Comment by Jeltr0n** [comment]
My project from 1.2.1 relies very heavily on the player sprite remaining the same between scenes. It seems like it's more inconvenient to re-set the player sprite on every screen than it would be to just set it back to what you want after a menu screen.

**Comment by chrismaltby** [comment]
This is matching the previous behaviour in 1.2.1 but I'd have to agree I had similar issues making the new sample project where I wished sometimes it worked the other way especially when switching between genres.

I could maybe add a checkbox like we have now on the "Attach Script" event to choose which behaviour to use

<img width="357" alt="Screenshot 2020-08-24 at 19 04 34" src="https://user-images.githubusercontent.com/16776042/91080049-d2f63b00-e63c-11ea-9649-f548db79fb20.png">

**Comment by Jeltr0n** [comment]
That sounds great! Would it have to store the old sprite sheet or would it revert to the starting sprite sheet when restoring it though?

**Comment by m-masaa** [comment]
I think my issue is mostly when restoring a previous scene from the stack. In that case, wouldn't it make sense to also revert the player character to the previous sheet?

> I could maybe add a checkbox like we have now on the "Attach Script" event to choose which behaviour to use
This at least sounds good to me for normal scene changes.

**Comment by Jeltr0n** [comment]
I agree because although I use the persistent version for when a character changes 'jobs' in my game, it is pretty jarring to come back from the menu as a cursor. I had to make a pretty specific custom script for restoring the player sprite that would be unnecessary if it was how gbstudio handled it.... but I don't know all of the things people use the stack for. It might mess it up for someone who uses it for more than just menus?

**Comment by chrismaltby** [comment]
I've just put a PR for this up at #481 I'm just going to hold off merging until I've done some more tests but I think this should work
